# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1295433757 18000
#      Wed Jan 19 05:42:37 2011 -0500
# Node ID de5fba8337c58443a87b8bf5156060af7de3ba2d
# Parent  5c18da80e28a9ca3cffe06727cd906a82870a513
octave_value_list::all_scalars: test for scalars, not strings

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-19  John W. Eaton  <jwe@octave.org>
+
+	* oct-obj.cc (octave_value_list::all_scalars): Check for scalar
+	values, not strings.  Bug #32172.
+
 2011-01-17  Rik  <octave@nomad.inbox5.com>
 
 	* DLD-FUNCTIONS/module-files: Add quadcc.cc to list of files.
 	* DLD-FUNCTIONS/quad.cc: Add Seealso links to quadcc.
 
 2011-01-17  Jaroslav Hajek  <highegg@gmail.com>
 
 	* DLD-FUNCTION/lookup.cc (Flookup): Validate option string.
diff --git a/src/oct-obj.cc b/src/oct-obj.cc
--- a/src/oct-obj.cc
+++ b/src/oct-obj.cc
@@ -174,21 +174,26 @@ octave_value_list::all_strings_p (void) 
       return false;
 
   return true;
 }
 
 bool
 octave_value_list::all_scalars (void) const
 {
-  octave_idx_type n = length (), i;
+  octave_idx_type n = length ();
 
-  for (i = 0; i < n && elem (i).is_string (); i++) ;
+  for (octave_idx_type i = 0; i < n; i++)
+    {
+      dim_vector dv = elem(i).dims ();
+      if (! dv.all_ones ())
+        return false;
+    }
   
-  return i == n;
+  return true;
 }
 
 bool
 octave_value_list::any_cell (void) const
 {
   octave_idx_type n = length ();
 
   for (octave_idx_type i = 0; i < n; i++)
